### PR TITLE
feat: add SubmitButton component to show loading state while submitting

### LINKED ISSUE #199: Duplicate lead creation on form submissions
**Describe the bug**

@monetmignon reported multiple instances of duplicate lead creation in the CRM.

Can confirm for the Google Sheet integrations, there's a case of duplicate records:
https://docs.google.com/spreadsheets/d/14SWi7kQDECm_xL0GutHvmYczv5wHtfeIaEILBKOZ-cM/edit#gid=166946480

**Expected behavior**

Each form submission should only create 1 record.

**Desktop (please complete the following information):**

Unknown

**Smartphone (please complete the following information):**

Unknown

### PR DESCRIPTION
**Pull request by nhsz**
This PR adds loading state to submit button and disables the button while submitting, to improve UX and avoid duplicate leads creation 

## Description

- adds `SubmitButton` component w/loading state when submitting
- updates all the forms to use `SubmitButton`

## Related Issue

Closes #199

### CODE REVIEW

File: src/components/SubmitButton.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      <MotionBox
+        backgroundColor='brand.button.shadow'
+        h={height}
+        w={width}
+        position='absolute'
+        animate={shadowBoxControl}
+        opacity={!isValid ? 0 : 1}
+      />
+
+      <MotionButton
```
Comment (pettinarip) [inline src/components/SubmitButton.tsx:46]: I thought you were going to use https://chakra-ui.com/docs/components/form/button#button-loading-state
or does it make things harder?

File: src/components/forms/AcademicGrantsForm.tsx [hunk 2]
```diff
@@ -44,12 +37,10 @@ import { ACADEMIC_GRANTS_THANK_YOU_PAGE_URL, TOAST_OPTIONS } from '../../constan
 
 import { AcademicGrantsFormData, ApplyingAs, GrantsReferralSource } from '../../types';
 
-const MotionBox = motion<BoxProps>(Box);
-const MotionButton = motion<ButtonProps>(Button);
-
 export const AcademicGrantsForm: FC = () => {
   const router = useRouter();
   const toast = useToast();
+  const [isSubmitting, setIsSubmitting] = useState(false);
```
Comment (pettinarip) [inline src/components/forms/AcademicGrantsForm.tsx:43]: `useForm` gives you this state already https://react-hook-form.com/api/useform/formstate

You should pass a Promise (turn `onSubmit` to async) to `handleSubmit` instead.

**Review by pettinarip — COMMENTED** [review]
Nice job @nhsz. I left a couple of comments that could improve the overall implementation.

File: src/components/SubmitButton.tsx (hunk 1 above)
Comment (nhsz) [reply src/components/SubmitButton.tsx:46]: It doesn't allow to apply custom styles, e.g.: it resets the font family

File: src/components/forms/AcademicGrantsForm.tsx (hunk 2 above)
Comment (nhsz) [reply src/components/forms/AcademicGrantsForm.tsx]: much cleaner! code updated

**Review by pettinarip — APPROVED** [review]
Nice!

**Review by pettinarip — APPROVED** [review]

**Pull request merged**